tests/foreman/api/test_contentview.py

```diff
@@ -1984,17 +1762,15 @@ def test_positive_publish_promote_with_RH_ostree_and_other(self):
         # Sync repository
         rpm_repo = entities.Repository(id=repo_id)
         rpm_repo.sync()
-        content_view = entities.ContentView(organization=self.org).create()
         content_view.repository = [self.repo, rpm_repo]
         content_view = content_view.update(['repository'])
-        self.assertEqual(len(content_view.repository), 2)
+        assert len(content_view.repository) == 2
         content_view.publish()
-        lce = entities.LifecycleEnvironment(organization=self.org).create()
-        promote(content_view.read().version[0], lce.id)
-        self.assertEqual(len(content_view.read().version[0].read().environment), 2)
+        promote(content_view.read().version[0], self.module_lce.id)
```

Comment:
I did not understand how self.module_lce.id comes here. I thought that it is coming from the setup, but I did not catch that.

Reply:
Oops, That's not ```self.module_lce.id``` but its ```module_lce.id```, now fixed.

Those tests didnt due to skip bugzilla, and hence didnt catch that. Good catch !

Reply:
I was wondering where that is coming! These fixtures are doing the hidden task and got confused!